xbmc/filesystem/CurlFile.cpp

```diff
@@ -925,6 +940,16 @@ void CCurlFile::Reset()
   m_state->m_cancelled = false;
 }
 
+void CCurlFile::SetProxy(ProxyType type, const std::string &host,
```

Comment:
you don't need it at the other place if you do it here. even if you have to do it more than once, encapsulation is more important.